### PR TITLE
Modify shell scripts to pass shellcheck --severity=warning

### PR DESCRIPTION
**Pull request by johnbley**
Got all the shell scripts to pass `shellcheck --severity=warning` and added a CI step to ensure it stays that way.
The fixes were mostly the usual quoting rules. One additional change was that I modified the `instrument` scripts to `exec` the original script rather than run it as a subprocess.  This is the standard way we do it in [the lambda instrumentation](https://github.com/open-telemetry/opentelemetry-lambda/blob/a214e7d38eeedae74f69ddbfb7dad54e710a9225/nodejs/packages/layer/scripts/otel-handler#L15)

### CODE REVIEW

**Comment by johnbley** [comment]
Test failures appear unrelated: 
```
error NU1902: Warning As Error: Package 'OpenTelemetry.Instrumentation.Http' 1.8.0 has a known moderate severity vulnerability
```

File: build.sh [hunk 1]
```diff
@@ -29,7 +29,8 @@ function FirstJsonValue {
 
 # If dotnet CLI is installed globally and it matches requested version, use for execution
 if [ -x "$(command -v dotnet)" ] && dotnet --version &>/dev/null; then
-    export DOTNET_EXE="$(command -v dotnet)"
+    DOTNET_EXE="$(command -v dotnet)"
+    export DOTNET_EXE
 else
```
Comment (Kielek) [inline build.sh:34]: This files is part of template in https://github.com/nuke-build/nuke/blob/6eb779618e22b0cdf1fa27538b305f750b698d88/source/Nuke.GlobalTool/templates/build.sh

Could you please propagate changes there?

**Comment by Kielek** [comment]
> Test failures appear unrelated:
> 
> ```
> error NU1902: Warning As Error: Package 'OpenTelemetry.Instrumentation.Http' 1.8.0 has a known moderate severity vulnerability
> ```

Should be fixed by #3372 shortly.

**Comment by Kielek** [comment]
One more things. CI converts https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/script-templates/otel-dotnet-auto-install.sh.template to the final sh file by executing `BuildInstallationScripts` nuke steps. I think, this file should also be covered by the shellcheck.

**Comment by johnbley** [comment]
> One more things. CI converts https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/script-templates/otel-dotnet-auto-install.sh.template to the final sh file by executing `BuildInstallationScripts` nuke steps. I think, this file should also be covered by the shellcheck.

👍  I added `*.sh.template` to the CI step.

File: nuget/OpenTelemetry.AutoInstrumentation/contentFiles/any/any/instrument.sh [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 
-BASE_PATH="$(cd $(dirname "$0") && pwd)"
+BASE_PATH="$(cd "$(dirname "$0")" && pwd)"
 
 # Settings for .NET
 export ASPNETCORE_HOSTINGSTARTUPASSEMBLIES=OpenTelemetry.AutoInstrumentation.AspNetCoreBootstrapper
 export CORECLR_ENABLE_PROFILING=1
 export CORECLR_PROFILER="{918728DD-259F-4A6A-AC2B-B85E1B658318}"
-export CORECLR_PROFILER_PATH=$(ls ${BASE_PATH}/OpenTelemetry.AutoInstrumentation.Native.*)
+CORECLR_PROFILER_PATH="$(ls ${BASE_PATH}/OpenTelemetry.AutoInstrumentation.Native.*)"
```
Comment (pjanotti) [inline nuget/OpenTelemetry.AutoInstrumentation/contentFiles/any/any/instrument.sh:9]: This should work either way, I'm just curious why this change.

**Review by pjanotti — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by nrcventura — APPROVED** [review]

**Review by RassK — APPROVED** [review]
